pkg/dependency/parser/python/pyproject/pyproject.go

```diff
@@ -16,7 +20,26 @@ type Tool struct {
 }
 
 type Poetry struct {
-	Dependencies map[string]any `toml:"dependencies"`
+	Dependencies dependencies     `toml:"dependencies"`
+	Groups       map[string]Group `toml:"group"`
+}
+
+type Group struct {
+	Dependencies dependencies `toml:"dependencies"`
+}
+
+type dependencies map[string]struct{}
+
+func (d *dependencies) UnmarshalTOML(data any) error {
+	m, ok := data.(map[string]any)
+	if !ok {
+		return fmt.Errorf("dependencies must be map, but got: %T", data)
```

Comment:
nit: for consistency and stacktrace
```suggestion
		return xerrors.Errorf("dependencies must be map, but got: %T", data)
```

Reply:
Fixed https://github.com/aquasecurity/trivy/pull/8106/commits/d1e2450c7b1c887a54b3592e402af0861f1af121